schemas/Identifier.json

```diff
@@ -1,7 +1,7 @@
 {
   "$schema": "http://json-schema.org/draft-04/schema#",
   "title": "Identifier",
-  "description": "An identifier for a user or institution.",
+  "description": "An identifier for a user or institution. Between 2-20 characters inclusively. May include dashes (-), but not at the beginning or end and no more than one consecutively.",
```

Comment:
> Also, I might want to allow users to create account aliases to hand out to different people and I might not want the external party to know whether they've given their "real" account identifier or not.

Just realized that I was thinking of the usernames allowed during registration. You're thinking about usernames in general. I think we should use a strict regex for registration (along the lines of what this PR proposes), but a looser regex (allowing base64 and longer names) otherwise.

I'll update this PR to be less strict and add a separate set of checks to the ledger during registration.